### PR TITLE
TST: Parametrize JoinNode expansion tests over config ``needed_outputs``

### PR DESCRIPTION
**Pull request by oesteban**
Will help better test #2971.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/pipeline/engine/tests/test_join.py [hunk 1]
```diff
@@ -148,35 +142,47 @@ def _list_outputs(self):
         return outputs
 
 
-def test_join_expansion(tmpdir):
+@pytest.mark.parametrize('needed_outputs', [True, False])
```
Comment (effigies) [inline nipype/pipeline/engine/tests/test_join.py]:
```suggestion
@pytest.mark.parametrize('needed_outputs', ['true', 'false'])
```

File: nipype/pipeline/engine/tests/test_join.py [hunk 2]
```diff
@@ -148,35 +142,47 @@ def _list_outputs(self):
...
+    global _sum_operands
+    global _products
     tmpdir.chdir()
 
+    # Clean up, just in case some other test modified them
+    _products = []
+    _sum_operands = []
+    _sums = []
+
+    config.set('execution', 'remove_unnecessary_outputs', ['false', 'true'][needed_outputs])
```
Comment (effigies) [inline nipype/pipeline/engine/tests/test_join.py]:
```suggestion
    config.set('execution', 'remove_unnecessary_outputs', needed_outputs)
```

**Comment by oesteban** [comment]
I have tested locally in all python versions we are testing - since tests were already passing before the suggestions, I'll go ahead and merge to focus on the other PR and release some Circle/Travis resources.

**Comment by effigies** [comment]
Seems reasonable.

**Pull request merged**